examples/svm/plot_svm_kernels.py

```diff
@@ -1,93 +1,306 @@
+applied with a parameter `C=1`, which allows for a certain degree of
+misclassification tolerance.
 
+If the data is not linearly separable in the original feature space, a
+non-linear `kernel` parameter can be set. Depending on the kernel, the process
+involves adding new features or transforming existing features to enrich and
+potentially add meaning to the data. When a kernel other than `linear` is set,
+the SVC applies the kernel trick, which computes the similarity between pairs of
+data points using the kernel function without explicitly transforming the entire
+dataset. The kernel trick surpasses the otherwise necessary matrix
+transformation of the whole dataset by only considering the relations between
+all pairs of data points. The kernel function maps two vectors (each pair of
+observations) to their similarity using their dot product.
 
+The hyperplane can then be calculated using the kernel function as if the
+dataset were represented in a higher-dimensional space. Using a kernel function
+instead of an explicit matrix transformation improves performance, as the kernel
+function has a time complexity of :math:`O({n}^2)`, whereas matrix
+transformation scales according to the specific transformation being applied.
+
+In this example, we compare the most common kernel types of Support Vector
+Machines: the linear kernel (`linear`), the polynomial kernel (`poly`), the
+radial basis function kernel (`rbf`) and the sigmoid kernel (`sigmoid`).
 """
 
 # Code source: Gaël Varoquaux
 # License: BSD 3 clause
 
+# %%
+# Creating a dataset
+# ------------------
+# We create a two-dimensional dataset with 16 samples and two targets. We plot
+# the samples with the colors matching their respective targets.
 import matplotlib.pyplot as plt
 import numpy as np
 
-from sklearn import svm
+X = np.array(
+    [
+        [0.4, -0.7],
+        [-1.5, -1.0],
+        [-1.4, -0.9],
+        [-1.3, -1.2],
+        [-1.1, -0.2],
+        [-1.2, -0.4],
+        [-0.5, 1.2],
+        [-1.5, 2.1],
+        [1.0, 1.0],
+        [1.3, 0.8],
+        [1.2, 0.5],
+        [0.2, -2.0],
+        [0.5, -2.4],
+        [0.2, -2.3],
+        [0.0, -2.7],
+        [1.3, 2.1],
+    ]
+)
 
-# Our dataset and targets
-X = np.c_[
-    (0.4, -0.7),
-    (-1.5, -1),
-    (-1.4, -0.9),
-    (-1.3, -1.2),
-    (-1.1, -0.2),
-    (-1.2, -0.4),
-    (-0.5, 1.2),
-    (-1.5, 2.1),
-    (1, 1),
-    # --
-    (1.3, 0.8),
-    (1.2, 0.5),
-    (0.2, -2),
-    (0.5, -2.4),
-    (0.2, -2.3),
-    (0, -2.7),
-    (1.3, 2.1),
-].T
-Y = [0] * 8 + [1] * 8
-
-# figure number
-fignum = 1
-
-# fit the model
-for kernel in ("linear", "poly", "rbf"):
-    clf = svm.SVC(kernel=kernel, gamma=2)
-    clf.fit(X, Y)
+y = np.array([0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1])
 
-    # plot the line, the points, and the nearest vectors to the plane
-    plt.figure(fignum, figsize=(4, 3))
-    plt.clf()
+# Plotting settings
+fig, ax = plt.subplots(figsize=(4, 3))
+ax.grid(True)
```

Comment:
Let's remove the grid. It will be useless in the second part when we have the decision function